frontend/src/components/UserForm/UserForm.js

```diff
@@ -0,0 +1,198 @@
+import { useMutate } from "../../hooks/use-mutate"
+import { useRef, useState, useEffect} from "react"
+import { useUser } from "../../hooks/user-hooks"
+import { useParams } from "react-router"
+import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
+
+const UserForm = () => {
+  const formRef = useRef()
+  const { updateUser } = useMutate()
+  const { id } = useParams()
+  const user = useUser({ "id": id })
+  const isAdmin = user?.data?.data?.data.user.admin
+
+  const fetchedCountry = user?.data?.data?.data?.user.country;
+  const fetchedState = user?.data?.data?.data?.user.state;
+  const [country, setCountry] = useState("");
+  const [region, setRegion] = useState("");
+
+  useEffect(() => {
+    if (fetchedCountry) setCountry(fetchedCountry);
+    if (fetchedState) setRegion(fetchedState);
+  }, [fetchedCountry, fetchedState]);
+
+  if (user.isError) return <h1>Something went wrong!</h1>
+  if (user.isLoading) return <h1>Loading...</h1>
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    // Add the current country/state values to the form data
+    formRef.current.append("country", country);
+    formRef.current.append("state", region);
+
+    const formData = new FormData(formRef.current)
+    const data = Object.fromEntries(formData)
+    const adminValue = data.admin === 'true' ? true : false;
+    data.country = country;
+    data.state = region;
+    if (!id) {
+        console.log('No user found')
+    } else {
+      const input = {
+        id: id,
+        first_name: data.firstName,
+        middle_name: data.middleName,
+        last_name: data.lastName,
+        gender: data.gender,
+        email: data.email,
+        admin: adminValue,
+        address1: data.address1,
+        address2: data.address2,
+        city: data.city,
+        state: data.state,
+        country: data.country,
+        zipcode: data.zipcode,
+        phone_number: data.phoneNumber
+      }
+      updateUser.mutate(input)
+    }
+  }
+
+  return (
+    <>
+      <form ref={formRef} onSubmit={handleSubmit} className="max-w-2xl m-6">
+        <h1 className="text-2xl text-center mb-3">{id ? "Edit User" : "Add User"}</h1>
+        <label className="admin-form-label">
+          First Name:
+          <input
+            className="form-input admin-form-input"
+            defaultValue={user?.data?.data?.data?.user.firstName && user.data.data.data.user.firstName}
```

Comment:
Also of note but is there a reason for the conditional `&&` check here?
It is implied that user exists at this point in the code as we've done a check above with using-
```
  if (status === "error") return <h1>Something went wrong!</h1>
  if (status === "loading") return <h1>Loading...</h1>
```
This looks to be remnants of past code that made it's way here. I think this was used to ensure conditionally render users in the case this was a create event and not an update event. Since this is now an update event only, the conditional rendering is not necessary. This can be cleaned up simply by using -
```
defaultValue={user.firstName}
```